### PR TITLE
Feat/implement pass

### PR DESCRIPTION
**Pull request by Dlutermade**
1. check player have cards left
     if the player has no cards left and the discard card eq null
        - can Pass

 2. check player have the card
     get player hands if hands can find the discard card
        - can Pass
     else not find
        - Error

need: https://github.com/Game-as-a-Service/saboteur-remix/issues/73

### CODE REVIEW

**Comment by kayac-chang** [comment]
@Dlutermade please write testing before implementation

**Comment by Dlutermade** [comment]
@kayac-chang ok!
測試補好了!

File: apps/backend/src/action/logic/pass.ts [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
+export type PassFailedError = ReturnType<typeof PassFailedError>;
+
+export type PassError = RepositoryWriteError | PassFailedError;
 
 export interface Pass {
   (
     source: EventSource<TurnHasBeenPassedEvent>,
     command: PassCommand
-  ): ResultAsync<TurnHasBeenPassedEvent[], Error /* PassError */>;
+  ): ResultAsync<TurnHasBeenPassedEvent[], PassError>;
```
Comment (kayac-chang) [inline apps/backend/src/action/logic/pass.ts:23]: could you help me update this by 
```ts
export type Pass = (
  source: EventSource<Event>,
  command: PassCommand
) => ResultAsync<TurnHasBeenPassedEvent, PassError>;
```

Comment (kayac-chang) [reply apps/backend/src/action/logic/pass.ts:23]: The `Event` type could be found in `~/action/event`

File: apps/backend/src/action/logic/pass.ts [hunk 2]
```diff
@@ -1,21 +1,28 @@
-import { ResultAsync, errAsync } from "neverthrow";
+import { ResultAsync, errAsync, okAsync } from "neverthrow";
 import { EventSource } from "~/models/event";
 import { PassCommand } from "~/action/command";
 import { TurnHasBeenPassedEvent } from "~/action/event";
 import { always, error } from "~/utils";
 
+import { Event as ActionEvent } from "~/action/event";
+import { P, match } from "ts-pattern";
+import { pipe } from "fp-ts/function";
```
Comment (kayac-chang) [inline apps/backend/src/action/logic/pass.ts:9]: please help me remove unuse variables.

**Review by kayac-chang — CHANGES_REQUESTED** [review]

File: apps/backend/src/action/logic/pass.ts (hunk 1 above)
Comment (Dlutermade) [reply apps/backend/src/action/logic/pass.ts:23]: ok ~  !

**Review by kayac-chang — APPROVED** [review]
LGTM

**Pull request merged**